nucleus/common/jul-extensions/src/main/java/fish/payara/logging/jul/PayaraLogger.java

```diff
@@ -0,0 +1,521 @@
+        final Supplier<String> msgSupplier) {
+        final PayaraLoggingStatus status = getLoggingStatus();
+        if (!isLoggable(level, status)) {
+            return;
+        }
+        final boolean processNow = isReady(status);
+        if (processNow || isToQueue(status)) {
+            logOrQueue(processNow, level, sourceClass, sourceMethod, msgSupplier.get(), thrown);
+        }
+    }
+
+
+    @Override
+    public void logrb(final Level level, final String sourceClass, final String sourceMethod,
+        final ResourceBundle bundle, final String msg, final Object... params) {
+        final PayaraLoggingStatus status = getLoggingStatus();
+        if (!isLoggable(level, status)) {
+            return;
+        }
+        final boolean processNow = isReady(status);
+        if (processNow || isToQueue(status)) {
+            final LogRecord record = new EnhancedLogRecord(level, msg, false);
+            record.setLoggerName(getName());
+            record.setSourceClassName(sourceClass);
+            record.setSourceMethodName(sourceMethod);
+            record.setParameters(params);
+            if (bundle != null) {
+                record.setResourceBundleName(bundle.getBaseBundleName());
+                record.setResourceBundle(bundle);
+            }
+            logOrQueue(processNow, record);
+        }
+    }
+
+
+    @Override
+    public void logrb(final Level level, final String sourceClass, final String sourceMethod,
+        final ResourceBundle bundle, final String msg, final Throwable thrown) {
+        final PayaraLoggingStatus status = getLoggingStatus();
+        if (!isLoggable(level, status)) {
+            return;
+        }
+        final boolean processNow = isReady(status);
+        if (processNow || isToQueue(status)) {
+            final LogRecord record = new EnhancedLogRecord(level, msg, false);
+            record.setLoggerName(getName());
+            record.setSourceClassName(sourceClass);
+            record.setSourceMethodName(sourceMethod);
+            record.setThrown(thrown);
+            if (bundle != null) {
+                record.setResourceBundleName(bundle.getBaseBundleName());
+                record.setResourceBundle(bundle);
+            }
+            logOrQueue(processNow, record);
+        }
+    }
+
+
+    @Override
+    public void entering(final String sourceClass, final String sourceMethod, final Object params[]) {
+        final PayaraLoggingStatus status = getLoggingStatus();
+        if (!isLoggable(Level.FINER, status)) {
+            return;
+        }
+        String msg = "ENTRY";
+        if (params == null) {
+            logp(Level.FINER, sourceClass, sourceMethod, msg);
+            return;
+        }
+        final boolean processNow = isReady(status);
+        if (!processNow && !isToQueue(status)) {
+            return;
+        }
+        for (int i = 0; i < params.length; i++) {
+            msg = msg + " {" + i + "}";
```

Comment:
String concatenation in a loop (but probably only short loops)

Reply:
Same in java.util.Logger, sadly it cannot be reused, so I had to copy paste it. But yeah, probably it can be optimized a bit.
(we can also change the format, maybe later; this method is not use much, because it's output is ugly)